src/main/java/org/reaktivity/nukleus/socks/internal/stream/AcceptTransitionListener.java

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright 2016-2017 The Reaktivity Project
+ *
+ * The Reaktivity Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package org.reaktivity.nukleus.socks.internal.stream;
+
+import java.util.Optional;
+
+public interface AcceptTransitionListener<T>
+{
+    void transitionToConnectionReady(Optional<T> connectionInfo);
```

Comment:
This seems like an anti pattern passing an Optional<T> around, why is it needed.  Seems like some of the classes that implement this don't even specify what extends T?

Reply:
I had some doubts on that also. The idea was to synchronize the 2 state machines Accept and ConnectReply. Here: https://github.com/reaktivity/nukleus-socks.java/pull/1/files/094695c075cf04b316c6e9b6c06924757e22e815#diff-830e54d6a783e80a6cbec2335eb4e894R87 I pass the TcpBeginEx.